samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js

```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
 const QNAMAKER_DIALOG = 'qnamaker-dialog';
 const QNAMAKER_BASE_DIALOG = 'qnamaker-base-dailog';
 
-class QnAMakerBaseDialog extends ComponentDialog {
+class QnAMakerBaseDialog extends QnAMakerDialog {
     /**
      * Core logic of QnA Maker dialog.
      * @param {QnAMaker} qnaService A QnAMaker service object.
      */
-    constructor(qnaService) {
-        super(QNAMAKER_BASE_DIALOG);
-
-        this._qnaMakerService = qnaService;
-
-        this.addDialog(new WaterfallDialog(QNAMAKER_DIALOG, [
-            this.callGenerateAnswerAsync.bind(this),
-            this.callTrain.bind(this),
-            this.checkForMultiTurnPrompt.bind(this),
-            this.displayQnAResult.bind(this)
-        ]));
-
-        this.initialDialogId = QNAMAKER_DIALOG;
+    constructor(knowledgebaseId, authkey, host) {
+        console.log('qnamkaerbasedialog'+ knowledgebaseId+' '+authkey+' '+host)
+        var noAnswer = ActivityFactory.DefaultNoAnswer;
+        var strictFilters = null;
+        super(knowledgebaseId, authkey, host, noAnswer, 0.3, 'Did you mean:','None of the above.',
+        3, ActivityFactory.cardNoMatchResponse, strictFilters, QNAMAKER_BASE_DIALOG);
+        this.id = QNAMAKER_BASE_DIALOG;
     }
 
-    /**
-    * @param {WaterfallStepContext} stepContext contextual information for the current step being executed.
-    */
-    async callGenerateAnswerAsync(stepContext) {
-        // Default QnAMakerOptions
-        var qnaMakerOptions = {
-            scoreThreshold: DefaultThreshold,
-            top: DefaultTopN,
-            context: {},
-            qnaId: -1
+    async GetQnAMakerOptions(dc)
+    {
+        console.log('GetQnAMakerOptions')
+        return new QnAMakerOptions
+        {
+            ScoreThreshold = DefaultThreshold,
+            Top = DefaultTopN,
+            QnAId = 0,
```

Comment:
Can you please verify that this function is overriding base class's function. 
I am not sure how javascript implements overrides but in base class' function definition, no where it is mentioned that this particular function can be overridden.